File: benchmarks/javascript/objects.js
#!/usr/bin/env bun

class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
    
    dist() {
        return this.x * this.x + this.y * this.y;
    }
}

let sumVal = 0;

for (let i = 0; i < 100000; i++) {
    const p = new Point(i, i + 1);
    sumVal += p.dist();
}

console.log(`Object result: ${sumVal}`);

File: benchmarks/python/objects.py
import time

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    
    def dist(self):
        return self.x * self.x + self.y * self.y

sum_val = 0

for i in range(100000):
    p = Point(i, i + 1)
    sum_val += p.dist()

print(f"Object result: {sum_val}")
